server/src/main/java/org/elasticsearch/tasks/TaskResultsService.java

```diff
@@ -70,7 +71,7 @@
 
     @Inject
     public TaskResultsService(Client client, ClusterService clusterService) {
-        this.client = client;
+        this.client = new OriginSettingClient(client, "tasks");
```

Comment:
`OriginSettingClient` removes a lot of the ceremony around this!

Reply:
Should this be `this.client = new OriginSettingClient(client, TASKS_ORIGIN`)?

Reply:
:+1: